Source/Core/DiscIO/Blob.h

```diff
@@ -74,6 +74,19 @@ class SectorReader : public IBlobReader
 	u64 m_cache_tags[CACHE_SIZE];
 };
 
+class CBlobBigEndianReader
+{
+public:
+	CBlobBigEndianReader(IBlobReader& reader) : m_reader(reader) {}
+
+	bool Read32(u64 offset, u32* buffer) const;
+	bool Read16(u64 offset, u16* buffer) const;
+	bool Read8(u64 offset, u8* buffer) const;
```

Comment:
I would prefer a templated read function instead:
```
template <typename T> bool Read(u64 offset, T* buffer) const {...}
```
But use whatever you like more.